src/components/Card.astro

```diff
@@ -1,11 +1,21 @@
 ---
 const { title, img, desc, url, badge, tags, target = '_blank' } = Astro.props;
+import {Picture} from '@astrojs/image/components'
+
 ---
 
 <div class="md:w-1/3 w-full">
   <a href={url} target={target}>
     <div class="card bg-base-100 transition ease-in-out hover:shadow-xl mx-6 my-2 hover:scale-[102%]">
-      <figure><img src={img} alt="Shoes" /></figure>
+      <figure>
+        <Picture 
+        aspectRatio="16:9"   
+        widths={[400, 800, 1200, 1800]} 
```

Comment:
Yes I agree I should've put more thoughts on these values , I sure we can come up with better ones

Reply:
how about these values:
   <Picture
            aspectRatio="16:9"
            widths={[1500,2500]}
            format={"webp"}
            sizes="(max-width: 1500px) 100vw, 2500px"
            src={heroImage}
            alt={title}
            class="w-full mb-6"
          />`
according to [squarespace ](https://support.squarespace.com/hc/en-us/articles/206542517-Formatting-your-images-for-display-on-the-web)we shouldn't go lower than 1500 (going lower will make images blurry) .
2500 should be fine for everything else
I've tested it on multiple devices 1500 works on mobile devices from 2014 and sometimes older so going lower doesn't make any sense

Reply:
To be fair this depends on the picture you are rendering, I think that size is overkill for the resolution we are rendering there. To get an idea of what's the actual size we are rendering the images, we can inspect the img tags in the browsers and see what's the resolution we have.

Take a look at one example.
![image](https://user-images.githubusercontent.com/53962116/232314652-2a4178a3-22ab-4515-8d3a-8c840d31c227.png)

the idea behind using this is try to fit the size of the image with the actual render size to save bandwidth and improve the page load

![image](https://user-images.githubusercontent.com/53962116/232314754-0efac8c6-c5ad-47a9-9ff2-f02780a93f3a.png)

![image](https://user-images.githubusercontent.com/53962116/232314777-659639e3-e3ef-4c01-b9a4-24094788f4ed.png)

Here you can see that this pic have different size renders for mobile or desktop layout, in this case for having the better performance we can use the different resolutions for the images. In any case this does not have to be 100% perfect, and we can use for example just one resolution a litter bigger that the actual render to avoid losing of quality,

Reply:
First I'd like to thank you for you patience and guidance I really appreciate it, so upon [further reading](https://victorzhou.com/blog/properly-size-images/) I think I get it now .
so how about these values x2 the render size to accommodate to retina displays [(ref)](https://www.danrodney.com/blog/retina-web-graphics-explained-1x-versus-2x-low-res-versus-hi-res/)
I tried to do something like this
```html
<Picture
              class="mask mask-circle"
              aspectRatio="1:1"
              widths={[152,304]}
              format={"webp"}
              sizes="(max-width: 750px) 100vw, 300px "
              src="/profile.jpg"
              alt="profile image"
            />
```
two issues:  
- it  always  picks the highest value and I couldn't get it to render the lower res photo  
- going with render value makes them blurry 
Overall I think these should be correct

Reply:
I'm taking a look at the render code and this is why is not working

![image](https://user-images.githubusercontent.com/53962116/232832067-6f3f5d6e-480c-48cf-97d0-a0e7ba31c346.png)

There is no media attr defined in the sources for the picture tag. 

It should have something like this 

https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_picture

I think that maybe this is due this is an experimental feature and maybe is not finished.

In this case, I think it should be better to use the Image instead Picture, at the end is not working as it should.

Reply:
Switched to images I hope they fix picture soon maybe we should report it 
ps  :for both images in store and post layout I added height since the lsp kept complaining but it compiles fine because aspectRatio and width are enough 
for the values I targeted x2 valueS like usual I hope this is enough